### ISSUE TITLE
deprecate twisted.python.util plugin functions

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#4533]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)| @exarkun reported|
|-|-|
|Trac ID|trac#4533|
|Type|task|
|Created|2010-06-30 12:40:41Z|

Left over from the old plugin system, `getPluginDirs` and `addPluginDir` don't do anything useful and should be removed.  First step, of course, is deprecation.

Attachments:

* [py-util-deprecate-4533.patch](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/3f6/3f6565c76837bcbe086fb5d262f18c232dee76e8/429c2a811bf1b7478e9b6e992222e2975e5592f3.patch) (3282 bytes) - added by thijs on 2012-05-14 12:44:47Z - 

<details><summary>Searchable metadata</summary>

```
trac-id__4533 4533
type__task task
reporter__exarkun exarkun
priority__low low
milestone__ 
branch__ 
branch_author__ 
status__closed closed
resolution__fixed fixed
component__core core
keywords__ 
time__1277901641000000 1277901641000000
changetime__1340126858000000 1340126858000000
version__None None
owner__thijs thijs
cc__thijs
```
</details>

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_2"></a>|@thijstriemstra commented|
|-|-|

Attached patch deprecates both functions, up for review. The .removal file would read:

```
addPluginDir and getPluginDirs in twisted.python.util are deprecated now.
```

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)<a name="note_3"></a>|@therve set owner to @thijstriemstra|
|-|-|

The patch should now say 12.2, but otherwise it looks good. Please merge.

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_4"></a>|@thijstriemstra set status to `closed`|
|-|-|

(In [34601]) Apply py-util-deprecate-4533.patch: addPluginDir and getPluginDirs in twisted.python.util are deprecated now.

Author: thijs
Reviewer: therve
Fixes: [#4533](https://github.com/twisted/twisted/issues/4533)

Left over from the old plugin system, getPluginDirs and addPluginDir don't do anything useful.
